src/renderer/components/sidebar-file-tree.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  Button,
+  ButtonGroup,
+  Classes,
+  Icon,
+  Menu,
+  MenuItem,
+  Tree,
+  TreeNodeInfo,
+} from '@blueprintjs/core';
+import { ContextMenu2 } from '@blueprintjs/popover2';
+import classNames from 'classnames';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { EditorId } from '../../interfaces';
+import { EditorPresence } from '../editor-mosaic';
+import { AppState } from '../state';
+
+interface FileTreeProps {
+  appState: AppState;
+}
+
+interface FileTreeState {
+  action: 'add' | 'default';
+}
+
+// crazy idea: maybe save state should be tied to the editors
+// and not to the filesystem
+
+@observer
+export class SidebarFileTree extends React.Component<
+  FileTreeProps,
+  FileTreeState
+> {
+  constructor(props: FileTreeProps) {
+    super(props);
+
+    this.state = {
+      action: 'default',
+    };
+  }
+
+  public render() {
+    const { editorMosaic } = this.props.appState;
+    const { files } = editorMosaic;
+
+    const fileList: TreeNodeInfo[] = Array.from(files)
+      .sort((a, b) => a[0].localeCompare(b[0]))
+      .map(([editorId, presence], index) => {
+        const visibilityIcon =
+          presence !== EditorPresence.Hidden ? 'eye-open' : 'eye-off';
+
+        return {
+          id: index,
+          hasCaret: false,
+          icon: 'document',
+          label: (
+            <ContextMenu2
+              content={
+                <Menu>
+                  <MenuItem
+                    icon="remove"
+                    text="Delete"
```

Comment:
This button should be disabled if the file is required (see `isRequiredFile()`).

If we try to run Electron on a fiddle where `main.js` has been removed, we're going to have a bad time :slightly_smiling_face:

Reply:
Addressed in d4d32a6